### DISCUSSION TITLE
General Uniform items

### DISCUSSION THREAD

**Discussion by Felderburg** [bcholmes/StarTrek2d20#245] [General]
Edit: Any images I provide here are under a CC0 license, as far as my contributions go.

I was going to comment on an STO uniform commit, but decided to make a general post. The STO uniforms look very cool, and I like the availability of captain and medical variants.

The Wrath of Khan uniforms have a different enlisted variant. The STO Odyssey's do as well: https://stowiki.net/wiki/Guide:_Uniform_colors/Odyssey_Uniforms In *theory*, it should be easier to implement, since it's literally a color swap between the shoulders, the chest stripe, and the chest. (https://stowiki.net/wiki/File:Odyssey_Officer_Service.png and https://stowiki.net/wiki/File:Odyssey_Enlisted_Service.png)

The Oddy uniforms also have a single black pip available, like the TNG ones did, but unless the TNG one gets added, no reason for the STO one to get added too.

The only category that has admiral uniforms is the DS9 variant; the TOS and ENT have flag rank stripes/bars but the uniforms don't vary. Otherwise, all the existing flag officer uniforms are not represented. (I only note this because the Oddy uniforms have flag variants.) Edit: Looks like flag was only added in November and is a current project; so never mind.

Cadet uniforms aren't available for any type (unless you count TWOK because cadets had regular rank insignia but the red trainee color).

Bug? TWOK has two blank enlisted uniform variants, that appear identical.

Edit after reading Patreon post: I don't think it's necessary to include every variant of the Oddy uniforms (or things like TNG admiralty uniforms, which have their own plethora of designs). Oddy alone has 'tactical' variants for every type, 2 flag versions, and a MACO division color.

Edited additions:

I would like to see things like alternate timelines on there (e.g. all good things/general future from 90s trek, or the future imperfect badges.

I think the DS9 and Dominion War badges have the bottom curve going too high. It's notably higher on the TNG badges, but gets lower for the "rectangle" backed badges.

I vaguely recall asking somewhere about contributing to uniforms, but it may have been to ships? Either way, is contributing art for uniforms something people can do?

**Comment by bcholmes** [comment]
I'm totally up for accepting art contributions.

**Reply by bcholmes** [reply]
A couple of things that are probably worth mentioning: 

1. All of the art assets are vector assets -- I receive most of them as an Illustrator file and convert them to .svg
2. Some assets -- including the uniforms, but also heads and the like -- have "magic colours" on them. By which I mean, there are specific colours that get substituted when I build the token. For example: on the uniforms, there's a red colour (#D30000) that I swap out in place of the division colours, where appropriate.
3. If you're going to contribute something, understand the free nature of this codebase's license: you'd need to provide the art assets as a [CC0](https://creativecommons.org/public-domain/cc0/)-like license. I'll credit you on the website, but all of this code is available in a way that allows others to come along and change it.

**Comment by Felderburg** [comment]
~~Adding this here, because the commits in my fork were bad and I want to rebase before comitting.~~ ~~Did the thing, sent a pull request.~~

As far as I can tell when looking at the files, this should work and be 400x400 like the current ones. However, when opening them in github, they appear to be visually different. The file I think should work is attached here; my KB-Test fork has exports at other DPIs.

<hr>
Providing this file under CC0. This is the enlisted version, and it *should* be properly sized and the insignia should work properly.

Per the STO guides, enlisted and officers have the same tops with color swaps, so the color swap is all that I did. Left shading shapes and badges the same.

Technically there might be a tiny amount of difference between the greys used for the officer shoulders and the enlisted main body (H22 vs H21); however, without STO open to see the RGB values, and because I don't think it matters visually for this usage, I did not modify that grey.

Also of note: enlisted have different pants, which is not at all relevant for these tokens, but might be worth noting in the event it ever becomes relevant.

![stoAverageFemaleEnlisted](https://github.com/user-attachments/assets/e53962e0-e160-45f0-bf47-6a1c440855c6)

**Reply by bcholmes** [reply]
I think there are a couple more differences. That border stripe between the shoulders and the main body is very different. It looks kind of rubbery and is segmented into an upper and lower border. The material seems to catch light a bit differently. Also, it doesn't look like there's piping between the collar and the upper-chest border: there's just the zipper. I'd be inclined to make it more like this:

![401266243-e53962e0-e160-45f0-bf47-6a1c440855c6](https://github.com/user-attachments/assets/83a2c569-3255-4284-adc7-1b0d06a9c5cf)

Otherwise, the 400x400 px sizing and SVG layers seem to have worked perfectly.

**Reply by bcholmes** [reply]
And then, awkwardly, there doesn't seem to be a good source for what enlisted pips look like.

**Reply by bcholmes** [reply]
It's tempting to throw in an indicator of the zipper, too.

![401266243-e53962e0-e160-45f0-bf47-6a1c440855c6](https://github.com/user-attachments/assets/6f8c41e3-7998-4c83-980e-433f4d5b68a7)

**Reply by Felderburg** [reply]
Well, the uniform top they use is the same, but they do indeed look different in the guide pictures. I'll have to log in and see how they look with consistent lighting. Because lighting in STO is... a Thing. But *in theory* it's only a color swap :p The uniform guide is 10 (!!!) years old at this point, so it's definitely worth seeing—the "Lighting 2.0" update was released 2 years *after* the guide was released.

**Reply by Felderburg** [reply]
> And then, awkwardly, there doesn't seem to be a good source for what enlisted pips look like.

In the guide pictures, it looks like they're all just wearing a single black pip, a la O'Brien's single round black pip (which was non-canonically called a "warrant officer" rank). I'm *preeeetty* sure there isn't a separate enlisted pip/rank style other than that for STO, but will also check that out in-game.

**Comment by Felderburg** [comment]
Ok, took some screenshots of the character creator. I have many more (and can generate more), but the first two demonstrate the officer and enlisted service uniforms in the "default lighting". I also included a comparison of the enlisted in all 4 lighting options (and can add one for officer if desired). The one enlisted pip is labelled as "chief" in the character creator, and is indeed a single black pip (like the LT JG or LT Cmdr final pips). The rest of the options are the Odyssey version of the rest of the TNG ranks, so no special O'Brien Chief pips.

I also took the reported RGB values, but given how different the uniform looks in the lighting options, I'm not sure it matters.

A1: 34 12 12
A6: 66 23 23
A8: 77 28 28
H21: 45 44 40
H22: 37 37 34
I1: 21 21 21

![crop_screenshot_2025-01-16-20-27-26](https://github.com/user-attachments/assets/ae7f5623-9521-4a84-9cb7-d19243912816)
![crop_screenshot_2025-01-16-20-29-46](https://github.com/user-attachments/assets/edfe5fd6-1ee4-4297-87bb-edf5d3e46493)![Comparison](https://github.com/user-attachments/assets/f55673ec-420b-4fb1-b172-18ef1255ff56)

**Reply by bcholmes** [reply]
Okay. I see what you're saying. I'm not 100% sure what I think of that. To me, it seems like the in-game experience is "simplifying" a bit by reusing the same model. 

(I kind of think the same thing about the rank insignia -- the single hollow pip doesn't really provide the specificity of the different enlisted ranks. STO rarely shows us enlisted folks, so it's probably not a priority for them. So, again, it just seems like a bit of a simplistic hand-wave).

**Reply by Felderburg** [reply]
Oh! Actually, they do show _loads_ of enlisted: the duty officers! I don't know know why I didn't think of it. The Duty Officer (DOff/doff) system is a way for you to send junior crew out on missions. They include petty officers and specialists, each of which do have rank icons (only represented in the DOff system—I don't recall any of them on costumes). Links at the bottom.

The big question is how much any of this matters for tokens. Try for 100% accuracy? Or a general verisimilitude to their look? (I lean to the latter; people can tell what they are, and tokens are somewhat abstractions of the "real" thing anyway.) I think STO's initial model, prior to DOffs being introduced, counted on people who cared just roleplaying their Bridge Officers (BOffs) as whatever rank they needed them to be. Since the BOffs were the only* aspect of crew that players saw at the time.

The "Rank" section within the overview is what I want to note here: https://stowiki.net/wiki/Duty_officer#Overview
Icons for petty officers in the chart here: https://stowiki.net/wiki/Specialization:_Fabrication_Engineer#Available_Fabrication_Engineers
Icons for specialists in the chart here: https://stowiki.net/wiki/Specialization:_Biologist#Available_Biologists
("Icons" because they're simply not present anywhere other than the border of DOff pictures.) Both charts have all factions, so check on the left whether they're Starfleet or not.

*There was at one time a crew 'count' on the space HUD, that would go up and down in combat. But it was especially narratively jarring to see hundreds of people die and then come back instantly, and didn't ever have an actual effect on the ship.

**Comment by Felderburg** [comment]
I've made flag uniforms per the guide (so tactical insignia on red only). They're the same, but one has silver and the other gold ('make new flags, but keep the old...'). Posting individual versions of each here, as well as a version that has both colors in layers (edit: posting here rather than pull requesting because I wasn't sure if you would prefer individual or layered). No rank pips. ECAF29 is the gold and D7D7D7 the silver are hex codes for the colors I used for rank pip matching if you get to them first. (I should probably get good screenshots to make sure they look right; I recall that they're like the TNG ones, regular pips in a box.)
Edit: probably could have made the silver and gold shoulder piping shinier.
![stoAverageMaleLowerFlag](https://github.com/user-attachments/assets/d1449fb5-7c6f-4e5f-a911-46556f5e0fc5)
![stoAverageMaleUpperFlag](https://github.com/user-attachments/assets/6f3711c2-3607-4d5c-8900-140e2298ef79)
![stoAverageMaleAllFlag](https://github.com/user-attachments/assets/ef5d5faa-0299-4490-95b4-5e366db3d28e)
![stoAverageFemaleLowerFlag](https://github.com/user-attachments/assets/aad8507a-ac38-45e1-a7ba-23b30d277944)
![stoAverageFemaleUpperFlag](https://github.com/user-attachments/assets/84634728-2836-426a-8922-ef10a0cb5468)
![stoAverageFemaleAllFlag](https://github.com/user-attachments/assets/837fd3e0-0c45-445c-a6ee-c657a9e61272)

**Reply by Felderburg** [reply]
With flag uniforms and service uniforms for officers and enlisted, as well as the captain and medical variants, these uniforms are as complete (if not more complete) than others. Dress, tactical/excursion, cadet versions of all 3, and potentially Intelligence and MACO variants are available variants for future expansion.

**Comment by Felderburg** [comment]
This is a 4-pip admiral and the 4-pip captain comparison. The flag officer pips are indeed rectangles in bigger boxes, but it looks like the inner pips might be slightly smaller than the regular ones? I did see that the Dominion War flag officer pips are just the regular ones with a box (or at least there was no change in size or position when I switched), which would be 100% the way to go if it's easier.

![cropscreenshot_2025-01-22-20-44-22](https://github.com/user-attachments/assets/078f36e0-8413-4410-85ab-c7a105488776)
![cropscreenshot_2025-01-22-20-44-29](https://github.com/user-attachments/assets/ad7e84f4-3c69-40d7-a59e-87aebd65dfd4)

**Comment by TakeshiYamato01** [comment]
Some nice work on all of these.

Looking forward to seeing the Picard uniforms added, too. :)

**Comment by TakeshiYamato01** [comment]
OK... providing images I took of the Picard Uniforms - 2399 and 2401. 2401 uses a different color palette, so I had to use a different shade of red. And while there is Admiral's rank insignia, forgot to include it.
![Star Trek Online - Picard 2399 Uniform](https://github.com/user-attachments/assets/161e7ffb-75d3-4361-ae23-dc773e14f25a)
![Star Trek Online - Picard 2401 Uniform](https://github.com/user-attachments/assets/aa397e87-5f18-4115-a860-77096b392f7f)

**Reply by TakeshiYamato01** [reply]
These were taken on my PS4, then downloaded to my Phone, transferred to a Private Discord Server then downloaded to my computer before being uploaded here.

Hope they help

**Reply by Felderburg** [reply]
Thanks! I suspect that whoever creates the Picard uniforms can use these to supplement on-screen references.

**Comment by Felderburg** [comment]
Starting a new "thread" in this discussion for TNG variants. The ones I can think of now are: dress uniforms, flag uniforms, seasons 1-2 standard, season 1-2 skant (and *maaaaybe* some of the variants caused by the switch per the Memory Alpha article, but that's... a bit much, and I would argue not the *intent* of the show).

https://memory-alpha.fandom.com/wiki/Starfleet_uniform_(2350s-2370s)

Attached to this post is an initial attempt at the "old" style, male variant. I'll work on the female later; when I do, I'll be looking at rank pip placement, to ensure they're in the same spot for both. No need to change pip style, though.

Edit: I looked at necks, it looks like they extend down far enough for these uniforms. If not, let me know. Also, the skant top is different enough in that it has more black on the front, even if the tokens don't show the t-shirt sleeves.
![tngoldAverageMale](https://github.com/user-attachments/assets/5e7db89b-1105-4d99-8265-a1c45ab75514)

**Reply by bcholmes** [reply]
That looks nice. In particular, I think you nailed the piping on the top. 

But at the risk of being a bit picky, I might be inclined to do a couple of tweaks. First up, I'd probably push up the black shoulder piece a bit more. The effect looks a bit different on different characters, but I feel like the amount of black between the lower red piping and the main body of the uniform is very small. 

I also think that the seam down the front of the suit is essential.

![early-tng-AverageMale](https://github.com/user-attachments/assets/01243c9e-3a45-46df-b256-bb60cc480230)

I haven't shown this on the updated figure, but the I also think I'd tweak the shoulders. There's no "seam" between the sleeve and the body, so I'd probably change the way that looks. 

A thing I'm less certain about: the fabric is a lot more body-hugging. I might think about whether or not I want that to be reflected in the body shape.

**Reply by Felderburg** [reply]
`I think you nailed the piping on the top.`

Thank you! I was especially proud of that part.

Yes, the material is definitely different, and the black should be higher. I'm working with/from existing uniforms, partially because it's easier, and partially because it ensures that anything I make fits existing other pieces for sure. I did make the outline a wee bit thinner, and reduced the shoulder lines from pure black to shadow as an attempt to make it look "smaller" or tighter. There's nothing that would be harmed if the body itself changed shape, right? The heads would still fit in the necks.

**Reply by Felderburg** [reply]
I had another go based on your feedback. I downloaded some necks, just to check that the lower collar of the uniform would work, but the raw downloaded files don't seem to actually match up with the current TNG uniform art. What am I missing?

![image](https://github.com/user-attachments/assets/e21f84bc-b72c-479e-8190-57646b9f446c)
![image](https://github.com/user-attachments/assets/97d26a43-a1ec-4135-a042-3461ff1aa8bc)

**Reply by bcholmes** [reply]
I like what you've done with that second look. I think there still needs to be some black line to demarcate the chest from the left arm, but I like it.

I did a big retooling of the necks a while ago -- the necks are now common across the different uniforms. There are two files -- `body/maleNeck.svg` and `body/femaleNeck.svg` that contains the current necks.

**Reply by Felderburg** [reply]
I think the SVGs attached are good enough for final, assuming they fit the necks. I've also attached a screensnip of the two overlaid; the pips from TNG kiiiiiind of fit, but I would recommend shrinking them a wee bit (despite the acanonical nature of the size change :p ).
![tngoldAverageMale_v3 2e](https://github.com/user-attachments/assets/ecfd2a28-518c-465b-a952-ff5aa9459cb8)
![tngoldAverageFemale_Ve](https://github.com/user-attachments/assets/c0209d2d-a4c3-47d9-9218-a65165bebdb2)
![image](https://github.com/user-attachments/assets/b1c5c3d4-4a56-47b1-a436-4f12eefacb09)

**Comment by TakeshiYamato01** [comment]
Gonna add another new 'thread' for another uniform type - Romulan Republic Uniforms, because whether it's 2E or Captain's Log, **_SOMEONE_** is gonna want to do a Romulan Republic game (I know I've got one in the works).

So, I'll start with the 'official' color guides:
![Star Trek Online Romulan Republic Uniform 1 - Officer](https://github.com/user-attachments/assets/de1db5b7-08b3-4f0d-bd33-dd8d18e845e9)
![Star Trek Online Romulan Republic Uniform 2 - Enlisted](https://github.com/user-attachments/assets/046ea236-46d0-415b-ad6e-76437b050a0f)
